File: app/src/main/java/com/alletsxlab/proguardstripslog/utils/LogUtil.java
package com.alletsxlab.proguardstripslog.utils;

import android.util.Log;

import com.alletsxlab.proguardstripslog.BuildConfig;

public class LogUtil {

    private String tag = "";
    private boolean isDebug = true;

    public LogUtil(String tag) {
        this.tag = tag;
    }

    public LogUtil(String tag, boolean isDebug) {
        this.tag = tag;
        this.isDebug = isDebug;
    }

    public LogUtil(Class<?> clz) {
        this.tag = clz.getSimpleName();
    }

    public LogUtil(Class<?> clz, boolean isDebug) {
        this.tag = clz.getSimpleName();
        this.isDebug = isDebug;
    }

    public LogUtil(Object obj) {
        this.tag = obj.getClass().getSimpleName();
    }

    public LogUtil(Object obj, boolean isDebug) {
        this.tag = obj.getClass().getSimpleName();
        this.isDebug = isDebug;
    }

    public boolean canLog() {
        return !BuildConfig.IS_RELEASE && isDebug;
    }

    public void d(String msg) {
        if (!canLog()) {
            return;
        }

        Log.d(tag, msg);
    }

    public void i(String msg) {
        if (!canLog()) {
            return;
        }

        Log.i(tag, msg);
    }

    public void w(String msg) {
        if (!canLog()) {
            return;
        }

        Log.w(tag, msg);
    }

    public void e(String msg) {
        if (!canLog()) {
            return;
        }

        Log.e(tag, msg);
    }

}

File: appkt/src/main/java/com/alletsxlab/proguardstripslog/utils/LogUtil.kt
package com.alletsxlab.proguardstripslog.utils

import android.util.Log
import com.alletsxlab.proguardstripslog.BuildConfig

class LogUtil(
    private val tag: String,
    private var isDebug: Boolean = true,
) {

    constructor(clz: Class<*>, isDebug: Boolean = true) : this(clz.simpleName, isDebug) {
    }

    constructor(obj: Any, isDebug: Boolean = true) : this(obj.javaClass.simpleName, isDebug) {
    }

    fun canLog(): Boolean {
        return !BuildConfig.IS_RELEASE && isDebug
    }

    fun d(msg: String) {
        if (!canLog()) {
            return
        }

        Log.d(tag, msg)
    }

    fun i(msg: String) {
        if (!canLog()) {
            return
        }

        Log.i(tag, msg)
    }

    fun w(msg: String) {
        if (!canLog()) {
            return
        }

        Log.w(tag, msg)
    }

    fun e(msg: String) {
        if (!canLog()) {
            return
        }

        Log.e(tag, msg)
    }

    fun d(format: String, vararg args: Any) {
        if (!canLog()) {
            return
        }

        Log.d(tag, String.format(format, *args))
    }

    fun i(format: String, vararg args: Any) {
        if (!canLog()) {
            return
        }

        Log.i(tag, String.format(format, *args))
    }

    fun w(format: String, vararg args: Any) {
        if (!canLog()) {
            return
        }

        Log.w(tag, String.format(format, *args))
    }

    fun e(format: String, vararg args: Any) {
        if (!canLog()) {
            return
        }

        Log.e(tag, String.format(format, *args))
    }

}